### PR TITLE
Docs concepts restructure

### PR DESCRIPTION
**Pull request by christinaausley**
## Description

Initial Concepts restructure for Identity concepts!

@Ben-Sheppard @dlavrenuek FINALLY I have gotten around to PRing this for the `next` version of docs! You'll notice I went through the existing Self-Managed Concepts and migrated short, tiny, Identity-specific details into their distinguished sections in the user guide.

For the remaining concepts in the access control section, would you like to see this content formatted in a different way? Are there concepts we don't have that relate to Self-Managed but should? Do you like the current structure?

This is of course open to change, so please don't hold back on feedback! Are there additional reviewers from the Self-Managed team you would like to have a look at this?

Closes https://github.com/camunda/developer-experience/issues/65.

## When should this change go live?

- [ ] This change is not yet live and should not be merged until {ADD_DATE} (apply `hold` label or convert to draft PR)?
- [x] There is no urgency with this change.
- [ ] This change or page is part of a marketing blog, conference talk, or something else on a schedule.
- [ ] This functionality is already available but undocumented.
- [ ] This is a bug fix or security concern.

## PR Checklist

- [ ] I have added changes to the relevant `/versioned_docs` directory, or they are not for an **already released version**.
- [x] I have added changes to the main `/docs` directory (aka `/next/`), or they are not for **future versions**.
- [x] My changes require an [Engineering review](https://github.com/camunda/camunda-platform-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned an engineering manager or tech lead as a reviewer, or my changes do not require an Engineering review.
- [x] My changes require a [technical writer review](https://github.com/camunda/camunda-platform-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned @christinaausley as a reviewer, or my changes do not require a technical writer review.

### CODE REVIEW

**Review by akeller — APPROVED** [review]
I really like this change/consolidation.

I only have questions or non-blocking, follow ups on broader Identity topics:

* Is "application" a Keycloak term? I see a 1:1 mapping here for our C8 components. Every component is an application. But Zeebe isn't in the screenshots, so does that not include Zeebe?
* Do we have recommendations for the application type for each of our C8 components? 
* @Ben-Sheppard communicated to me via DM wanting to move away from "resource-based" as in "resource-based auth", but what about "resource based permissions"?

**Comment by christinaausley** [comment]
@Ben-Sheppard Will you have time to take a peek at this this week? No worries if you have other priorities, but just in case you missed this.

**Comment by Ben-Sheppard** [comment]
Sorry @christinaausley for not looking at this before, I will put it on my list to review today

File: docs/self-managed/identity/user-guide/authorizations/generating-m2m-tokens.md [hunk 1]
```diff
@@ -5,11 +5,15 @@ sidebar_label: "Generating machine-to-machine (M2M) tokens"
 description: "In this guide, we'll show you how to generate your own machine-to-machine (M2M) tokens."
 ---
 
-In this guide, we'll show you how to generate your own **machine-to-machine (M2M)** tokens.
+A **machine-to-machine (M2M)** token is a token requested by one service so it can
+communicate with another service acting as itself.
+
+In [Identity](/self-managed/identity/what-is-identity.md), we provide the ability to assign permissions to
+an application. This functionality allows an application to perform the `client_credentials` flow to
+retrieve a JWT token with permissions. The token generated can then be used to communicate with other applications in the Camunda Platform without
```
Comment (Ben-Sheppard) [inline docs/self-managed/identity/user-guide/authorizations/generating-m2m-tokens.md:13]:
```suggestion
retrieve a JWT token with permissions. The token generated can then be used to communicate with other applications in Camunda 8 without
```

File: docs/self-managed/identity/user-guide/authorizations/generating-m2m-tokens.md [hunk 2]
```diff
@@ -5,11 +5,15 @@ sidebar_label: "Generating machine-to-machine (M2M) tokens"
 description: "In this guide, we'll show you how to generate your own machine-to-machine (M2M) tokens."
 ---
 
-In this guide, we'll show you how to generate your own **machine-to-machine (M2M)** tokens.
+A **machine-to-machine (M2M)** token is a token requested by one service so it can
+communicate with another service acting as itself.
+
+In [Identity](/self-managed/identity/what-is-identity.md), we provide the ability to assign permissions to
+an application. This functionality allows an application to perform the `client_credentials` flow to
```
Comment (Ben-Sheppard) [inline docs/self-managed/identity/user-guide/authorizations/generating-m2m-tokens.md:12]: 🔧 This functionality doesn't strictly allow the `client_credentials` flow but rather another setting on the application.

 I wonder if we could say something like
```
... we provide the ability to assign permissions to an application. This allows an application to perform requests to other applications without the need for user intervention.
```

File: docs/self-managed/identity/user-guide/roles/add-assign-role.md [hunk 3]
```diff
@@ -5,11 +5,12 @@ sidebar_label: "Add and assign a role"
 description: "Use Identity to create a role and assign a role to a user."
 ---
 
+To use [Identity](/self-managed/identity/what-is-identity.md), an account is required. We refer to this account as a user. Access to the components within the stack can be controlled by the roles assigned to a user.
```
Comment (Ben-Sheppard) [inline docs/self-managed/identity/user-guide/roles/add-assign-role.md:8]:
```suggestion
To use [Identity](/self-managed/identity/what-is-identity.md), an account is required. We refer to this account as a user. Access to the components within Camunda 8 can be controlled by the roles assigned to a user.
```

**Review by Ben-Sheppard — COMMENTED** [review]
Just a couple of comments, some would be good to address (the `Camunda 8` changes) but others are really just a suggestion that I'm fine with either way :)

File: docs/self-managed/identity/user-guide/authorizations/generating-m2m-tokens.md [hunk 4]
```diff
@@ -5,11 +5,15 @@ sidebar_label: "Generating machine-to-machine (M2M) tokens"
 description: "In this guide, we'll show you how to generate your own machine-to-machine (M2M) tokens."
 ---
 
-In this guide, we'll show you how to generate your own **machine-to-machine (M2M)** tokens.
+A **machine-to-machine (M2M)** token is a token requested by one service so it can
```
[the remark this answers is not in the document]
Comment (christinaausley) [reply docs/self-managed/identity/user-guide/authorizations/generating-m2m-tokens.md:8]: Adjusted in latest commit 👍

File: docs/self-managed/identity/user-guide/authorizations/generating-m2m-tokens.md (hunk 2 above)
Comment (christinaausley) [reply docs/self-managed/identity/user-guide/authorizations/generating-m2m-tokens.md]: Adjusted in latest commit 👍

File: docs/self-managed/identity/user-guide/authorizations/generating-m2m-tokens.md [hunk 5]
```diff
@@ -5,11 +5,14 @@ sidebar_label: "Generating machine-to-machine (M2M) tokens"
 description: "In this guide, we'll show you how to generate your own machine-to-machine (M2M) tokens."
 ---
 
-In this guide, we'll show you how to generate your own **machine-to-machine (M2M)** tokens.
+A **machine-to-machine (M2M)** token is a token requested by one application so it can
+communicate with another application acting as itself.
+
+In [Identity](/self-managed/identity/what-is-identity.md), we provide the ability to assign permissions to an application. This allows an application to perform requests to other applications without the need for user intervention. This functionality allows an application to perform the `client_credentials` flow to
+retrieve a JWT token with permissions, for example. The token generated can then be used to communicate with other applications in Camunda 8 without
```
Comment (Ben-Sheppard) [inline docs/self-managed/identity/user-guide/authorizations/generating-m2m-tokens.md:12]: The following sentence can be removed 
```
This functionality allows an application to perform the `client_credentials` flow to
retrieve a JWT token with permissions, for example.
```

**Review by Ben-Sheppard — COMMENTED** [review]
Just one tweak from my side otherwise the content looks good to me :)

**Review by Ben-Sheppard — APPROVED** [review]
Thank you for addressing my comments :D

**Comment by akeller** [comment]
> I really like this change/consolidation.
> 
> I only have questions or non-blocking, follow ups on broader Identity topics:
> 
> * Is "application" a Keycloak term? I see a 1:1 mapping here for our C8 components. Every component is an application. But Zeebe isn't in the screenshots, so does that not include Zeebe?
> * Do we have recommendations for the application type for each of our C8 components?
> * @Ben-Sheppard communicated to me via DM wanting to move away from "resource-based" as in "resource-based auth", but what about "resource based permissions"?

@christinaausley I'll convert this to an issue and add it to the docs backlog for follow-up, but let's get this merged!

**Pull request merged**